pkg/iam/iam.go

```diff
@@ -68,18 +60,30 @@ func AddUser(log logr.Logger, session *session.Session, config AddUserConfig, us
 	return err
 }
 
-func SetAWSPolicy(log logr.Logger, credentials *credentials.Credentials, config AddUserConfig, userID string) error {
-	// AWS Config Object to create a session
-	awsConfig := &aws.Config{
-		Region:      aws.String(config.Region),
-		Credentials: credentials,
-	}
+func RemoveUser(client *Client, awsLoginRoles []string, username string) error {
 
-	// Initialize session to AWS
-	sess, err := session.NewSession(awsConfig)
+	policies, err := client.ListPolicies()
 	if err != nil {
-		return fmt.Errorf("session initialization for region %s: %w", config.Region, err)
+		return err
+	}
+
+	for _, policy := range policies {
+		if policy.Document.Exists(username) {
+			policy.Document.Remove(username)
+
+			if policy.Document.Count() == 0 {
+				err := client.DeleteAndDetachPolicy(policy, awsLoginRoles)
+				if err != nil {
+					return err
+				}
+			} else {
+				err := client.UpdatePolicy(policy)
+				if err != nil {
+					return err
+				}
+			}
+		}
```

Comment:
Another case of deep nesting. WDYT of this alternative?

```suggestion
		if !policy.Document.Exists(username) {
			continue
		}

		policy.Document.Remove(username)

		if policy.Document.Count() == 0 {
			err := client.DeleteAndDetachPolicy(policy, awsLoginRoles)
			if err != nil {
				return err
			}
			continue
		}
		err := client.UpdatePolicy(policy)
		if err != nil {
			return err
		}
```

Reply:
I generally agree with your preference for avoiding nesting but in this case I think the logic becomes less clear when the code is structured this way. The branching sort of supports the logical structure and it helps me read it quicker. If your answer is the idiomatic way to do it in go I would probably have felt differently if I had been coding go for longer. Let's go full-nerd on this on tomorrow's check-in, would love to hear more details and perspectives on this.

Reply:
Reduced nesting is somewhat idiomatic, but so is reading. I can see what you mean about the story the current form tells. 👍  Let's keep it as is 🚀